### PR TITLE
get / edit WAF package

### PR DESCRIPTION
**Pull request by sbfaulkner**
This PR adds entrypoints to get and edit WAF rule packages as per https://api.cloudflare.com/#waf-rule-packages-firewall-package-details and https://api.cloudflare.com/#waf-rule-packages-edit-firewall-package

It will contribute to implementation of #321

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by sbfaulkner** [comment]
@patryk @jacobbednarz would appreciate you taking a look ... I'm creating a few PRs to update the terraform provider

File: waf_test.go [hunk 1]
```diff
@@ -67,6 +67,109 @@ func TestListWAFPackages(t *testing.T) {
...
+		ActionMode:    "",
+	}
+
+	d, err := client.WAFPackage(testZoneID, "a25a9a7e9c00afc1fb2e0245519d725b")
+
+	if assert.NoError(t, err) {
+		assert.Equal(t, want, d)
+	}
+
+	_, err = client.WAFPackage(testZoneID, "123")
```
Comment (jacobbednarz) [inline waf_test.go]: I'm not generally a fan of multiple assertions in a single test. Instead, explicit and isolated tests are my preferred option to ensure we can use specific test names and don't end up with a single test with lots of responsibilities. Not a hill I'm willing to die on here though :)

Comment (sbfaulkner) [reply waf_test.go]: seemed to be a common pattern in other similar tests -- there's probably several things we could do to clean up the WAF-related (and other) tests, but probably not in scope for this PR

Comment (sbfaulkner) [reply waf_test.go]: eg. https://github.com/cloudflare/cloudflare-go/pull/323/files#diff-caef407880318468a47a25739545707eR66

**Comment by patryk** [comment]
It's great, thanks for the contribution @sbfaulkner. I'm sorry about the delay, I must've miss the notification.

**Pull request merged**